IN-688 Remove clusterings engine

adessy opened the pull request:
- IN-688
- [citizenlab-ee PR](https://github.com/CitizenLabDotCo/citizenlab-ee/pull/148)

adessy committed:
IN-688 Uninstall the `clusterings` engine

adessy committed:
IN-688 Drop clusterings table

adessy committed:
IN-688 Remove clusterings engine

adessy committed:
IN-688 Remove NLP::ClusteringService (dead code)

adessy committed:
Merge branch 'master' into IN-688/remove-clustering-engine

adessy committed:
IN-688 Drop clusterings table (2): run migration

IvaKop committed:
Remove tagging FE

Commit:
Translations updated by CI (extract-intl)

IvaKop committed:
Remove d3-hierarchy

IvaKop committed:
Merge branch 'IN-688/remove-clustering-engine' of github.com:CitizenLabDotCo/citizenlab into IN-688/remove-clustering-engine

IvaKop committed:
Remove unsued services

IvaKop commented on front/app/modules/commercial/user_custom_fields/services/stats.ts:
```diff
@@ -121,28 +121,3 @@ export function usersByDomicileStream(
   });
 }
-
-export function votesByGenderStream(streamParams: IStreamParams | null = null) {
```
@adessy heads up! These endpoints are not used anymore by the FE - should we delete them?

adessy replied:
Is it related to the clustering engine? Otherwise, I'd rather log this into Jira for a future PR.

IvaKop replied:
It was used in the clustering module on the FE only (got picked up by our dead code detection) but I don't have enough context to know whether it's strictly related 🤷

jinjagit replied:
Yes, my first thought was that this does not look obviously linked to whatever the clusterings engine was (though it might be).

IvaKop committed:
Remove unsued types

IvaKop committed:
Merge master

jinjagit commented:
@Adrien Is there any background info. available for this PR. Some info. that might help me:
- What is a clustering engine?
- Why are we removing it?
- Are there any concerns around removing it?

adessy commented:
@jinjagit Hey Simon! Sorry about that, I should have been more explicit.

Nope, there is not much context for this one. This is just an deprecated half-broken feature that we are removing. You could try to enable it on a platform to see what it does. This should add a dashboard panel somewhere in the back-office, I think. But chances are, it would not work.

jinjagit commented on front/app/translations/admin/ar-SA.json:
```diff
@@ -634,7 +590,6 @@
   "app.containers.AdminPage.DashboardPage.selectedProject": "عامل تصفية المشروع الحالي",
   "app.containers.AdminPage.DashboardPage.selectedTopic": "عامل تصفية الموضوع الحالي",
   "app.containers.AdminPage.DashboardPage.subtitleDashboard": "اكتشف ما يحدث على منصتك.",
-  "app.containers.AdminPage.DashboardPage.tabInsights": "الرؤى (بيتا)",
```
Is this definitely related to clusterings engine?

IvaKop replied:
It is! It's the name of the tab where the clustering module was displayed. :)

jinjagit commented on front/app/translations/admin/da-DK.json:
```diff
@@ -634,7 +590,6 @@
   "app.containers.AdminPage.DashboardPage.selectedProject": "nuværende projektfilter",
   "app.containers.AdminPage.DashboardPage.selectedTopic": "aktuelt tag-filter",
   "app.containers.AdminPage.DashboardPage.subtitleDashboard": "Find ud af, hvad der sker på din platform.",
-  "app.containers.AdminPage.DashboardPage.tabInsights": "Indsigt",
```
(Again) Is this definitely related to clusterings engine?

jinjagit commented on front/app/translations/admin/de-DE.json:
```diff
@@ -634,7 +590,6 @@
   "app.containers.AdminPage.DashboardPage.selectedProject": "aktueller Projektfilter",
   "app.containers.AdminPage.DashboardPage.selectedTopic": "aktueller Themenfilter",
   "app.containers.AdminPage.DashboardPage.subtitleDashboard": "Entdecken Sie, was auf Ihrer Plattform passiert.",
-  "app.containers.AdminPage.DashboardPage.tabInsights": "Einblicke",
```
(Again) Is this definitely related to clusterings engine?

jinjagit commented on front/app/translations/admin/en-CA.json:
```diff
@@ -634,7 +590,6 @@
   "app.containers.AdminPage.DashboardPage.selectedProject": "current project filter",
   "app.containers.AdminPage.DashboardPage.selectedTopic": "current tag filter",
   "app.containers.AdminPage.DashboardPage.subtitleDashboard": "Discover what's happening on your platform.",
-  "app.containers.AdminPage.DashboardPage.tabInsights": "Insights",
```
I'm guessing you're pretty confident about this one, so I'll stop commenting on every occurrence ;-)

IvaKop replied:
These changes happen automatically for all languages once a string is no longer needed ;)

jinjagit reviewed (approved):
I can't see any obvious problems.

I highlighted a line which is removed from many translations, but it seems you have selected it for a reason.

adessy commented:
Thanks for the review @Simon !  🙏

IvaKop committed:
Merge master

IvaKop commented:
@luucvanderzee Do you mind taking a quick look at the front-end part for this one? 
There's nothing special here, we are just deleting one of the deprecated modules, but it's always nice to have a second pair of eyes just in case 👀

adessy committed:
IN-688 Merge branch 'master' into IN-688/remove-clustering-engine

# Conflicts:
#	back/db/schema.rb
#	back/engines/commercial/clusterings/spec/acceptance/clusterings_spec.rb
#	back/engines/commercial/nlp/spec/services/clustering_service_spec.rb

adessy committed:
IN-688 Fix conflicts with master in schema.rb

adessy committed:
IN-688 Merge remote-tracking branch 'origin/IN-688/remove-clustering-engine' into IN-688/remove-clustering-engine

# Conflicts:
#	front/app/modules/commercial/clusterings/resources/GetClusterings.tsx
